src/Sqs/SqsClient.php

```diff
@@ -105,7 +105,68 @@ private function queueUrl()
     }
 
     /**
-     * Validates ReceiveMessage body MD5s
+     * Constructs the expected md5 hash of message attributes according to the encoding
+     * scheme detailed in SQS documentation.
+     *
+     * @param array $message Message containing attributes for validation.
+     *                       Retrieved when using MessageAttributeNames on
+     *                       ReceiveMessage.
+     *
+     * @return string|null The md5 hash of the message attributes according to
+     *                     the encoding scheme. Returns null when there are no
+     *                     attributes.
+     * @link http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-attributes.html#message-attributes-items-validation
+     */
+    public static function getExpectedMessageAttributesMd5($message)
```

Comment:
Maybe a better naming for this helper method? like `computeMessageAttributesMd5` or perhaps `caculateMessageAttributesMd5` or perhaps `messageAttributesMd5`? `get` feels like fetching value from response instead of constructing it. Perhaps to echo this for `MD5OfBody`, you could have another function `bodyMd5(..)` that simply returns `md5($msg['Body']` when available.